apps/04TowersOfHanoi.js

```diff
@@ -13,40 +13,101 @@ var stacks = {
   c: []
 };
 
+var winner = false;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+//cheat code
+	if(startStack === "cheat" & endStack === "code"){
+		stacks.a= [];
+		stacks.b = [];
+		stacks.c = [4,3,2,1];
+		printStacks();
+		return true;
+	}
+
+//check to see if the move is legal
+ 	if(!isLegal(startStack, endStack)){
+		return false;
+  	}
+//if it hasnt returned false, move the number and return to the main funcion
+	stacks[endStack].push(stacks[startStack].pop());
+	return true;
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+//check to see if the end stack is empty stack because, when we do length-1 to get the value, an empty stack will return undefined
+	if(stacks[endStack].length === 0)
+ 	{
+	 	return true;
+  	}
+//this is where we check to make sure the endstack has a value bigger than the start stack pop
+	if (stacks[startStack][(stacks[startStack].length-1)]<stacks[endStack][(stacks[endStack].length-1)])
+	{
+	  return true;
+  	}
+//if we haven't returned true, it violates, so return false
+  return false;
 }
 
-function checkForWin() {
-  // Your code here
-
+function checkForWin(){
+	//after we make the first move, anytime we get back to four in a stack, thats a winner
+	if(stacks.a.length === 4 || stacks.b.length === 4 || stacks.c.length === 4){
```

Comment:
Would this allow a player to win in 2 moves? Placing "1" on an empty stack then back again on stack "a".